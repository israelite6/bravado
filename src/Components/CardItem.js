import React from 'react';
import PropTypes from 'prop-types';
import {View, StyleSheet} from 'react-native';
import HighlightText from '@sanar/react-native-highlight-text';
import FastImage from 'react-native-fast-image';

/**
 *
 * @param {string} email
 * @param {string} address
 * @param {string} name
 * @param {string} title
 * @param {string} avatar
 * @param {string} searchKeyword
 * @returns
 */

export default class CardItem extends React.PureComponent {
  render() {
    const {name, email, address, title, avatar, searchKeyword} = this.props;

    return (
      <View style={styles.root}>
        <View>
          <HighlightText
            style={styles.headerStyle}
            highlightStyle={styles.highlightStyle}
            searchWords={[searchKeyword || '']}
            textToHighlight={name}
          />
          <HighlightText
            style={styles.textStyle}
            highlightStyle={{backgroundColor: 'yellow'}}
            searchWords={[searchKeyword || '']}
            textToHighlight={email}
          />
          <HighlightText
            style={styles.textStyle}
            highlightStyle={{backgroundColor: 'yellow'}}
            searchWords={[searchKeyword || '']}
            textToHighlight={title}
          />
          <HighlightText
            style={styles.textStyle}
            highlightStyle={{backgroundColor: 'yellow'}}
            searchWords={[searchKeyword || '']}
            textToHighlight={address}
          />
        </View>
        <FastImage source={{uri: avatar}} style={styles.avatar} />
      </View>
    );
  }
}

CardItem.propTypes = {
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  searchKeyword: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: '#FFFFFF',
    paddingHorizontal: 8,
    paddingVertical: 7,
    marginTop: 15,
    borderRadius: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    shadowColor: 'rgba(0,0,0,0.1)',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.5,
    shadowRadius: 1,
    elevation: 1,
  },
  highlightStyle: {
    backgroundColor: '#FFFA86',
  },
  headerStyle: {
    fontWeight: '600',
    fontSize: 18,
  },
  textStyle: {
    fontWeight: '600',
    fontSize: 13,
    paddingTop: 5,
  },
  avatar: {
    backgroundColor: '#C4C4C4',
    borderRadius: 4,
    height: 68,
    width: 68,
    resizeMode: 'cover',
  },
});
