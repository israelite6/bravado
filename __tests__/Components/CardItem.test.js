/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CardItem from '../../src/Components/CardItem';

it('Avatar component renders correctly', () => {
  const tree = renderer
    .create(
      <CardItem
        name="name"
        email="email"
        address="address"
        title="title"
        avatar="avatar"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
