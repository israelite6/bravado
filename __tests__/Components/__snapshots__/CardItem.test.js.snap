// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 15,
      "paddingHorizontal": 8,
      "paddingVertical": 7,
      "shadowColor": "rgba(0,0,0,0.1)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      name
    </Text>
    <Text
      style={
        Object {
          "fontSize": 13,
          "fontWeight": "600",
          "paddingTop": 5,
        }
      }
    >
      email
    </Text>
    <Text
      style={
        Object {
          "fontSize": 13,
          "fontWeight": "600",
          "paddingTop": 5,
        }
      }
    >
      title
    </Text>
    <Text
      style={
        Object {
          "fontSize": 13,
          "fontWeight": "600",
          "paddingTop": 5,
        }
      }
    >
      address
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#C4C4C4",
          "borderRadius": 4,
          "height": 68,
          "resizeMode": "cover",
          "width": 68,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "avatar",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
